src/Compilers/Core/Portable/InternalUtilities/IComparableHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Roslyn.Utilities
+{
+    internal static class IComparableHelper
+    {
+        public static int CompareTo<T>(T first, T second, params Func<T, IComparable>[] comparingComponents)
```

Comment:
not certin we want this.  will this be double allocation (the function, and the array?).  Or do you intend to cache that all callside.